presto-spi/src/main/java/io/prestosql/spi/eventlistener/TracerEvent.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.eventlistener;
+
+import static java.util.Objects.requireNonNull;
+
+public class TracerEvent
+{
+    private final String nodeId;
+    private final String uri;
+    private final String queryId;
+    private final String stageId;
+    private final String taskId;
+    private final String pipelineId;
+    private final String splitId;
+    private final String operatorId;
```

Comment:
Use `Optional` for any field that may be omitted. It makes it easier for users of the API to understand its semantics.